src/prefect/engine/results/s3_result.py

```diff
@@ -163,9 +163,7 @@ def exists(self, location: str, **kwargs: Any) -> bool:
         import botocore
 
         try:
-            self.client.get_object(
-                Bucket=self.bucket, Key=location.format(**kwargs)
-            ).load()
+            self.client.get_object(Bucket=self.bucket, Key=location.format(**kwargs))
```

Comment:
dumb question but did you confirm that this will raise an error without the `load` call?

Reply:
Yeah I ran a flow using targets and the error was properly caught on the first run. Then the second run used the cache as expected 👍